go/vt/autoschema/validator.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2015, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package autoschema
+
+import (
+	"errors"
+
+	"github.com/youtube/vitess/go/vt/sqlparser"
+)
+
+// Error: Table does not exist
+var ErrUnknownTable = errors.New("Table does not exist")
+
+// Error: Too many rows
+var ErrTooManyRows = errors.New("Too many rows")
+
+// Error: Invalid Sql Statement
+var ErrInvalidSql = errors.New("Invalid Sql Statement")
+
+// Error: In Transaction at the end of sql statements
+var ErrInTransactionAtTheEnd = errors.New("In Transaction at the end of sql statements")
+
+// Validator apply a set of rules to perform on a schema change
+type Validator struct {
+}
+
+// TableSchema provides a view of MySQL table schema
+type TableSchema struct {
+	numRows int
+	columns []string
+}
+
+// Context carries any necessary information to help Validator apply rules
+type Context struct {
```

Comment:
Better to avoid this name since it has a very specific meaning in the entire system. I think you can get rid of the current stateless Validator, and call this one the Validator.